capsules/src/max17205.rs

```diff
@@ -68,35 +71,55 @@ enum State {
     ReadVolt,
     SetupReadCurrent,
     ReadCurrent,
+    SetupReadRomID,
+    ReadRomID,
 }
 
 pub trait MAX17205Client {
-    fn status(&self, status: u16);
-    fn state_of_charge(&self, percent: u16, capacity: u16, full_capacity: u16);
-    fn voltage_current(&self, voltage: u16, current: u16);
-    fn coulomb(&self, coulomb: u16);
+    fn status(&self, status: u16, error: ReturnCode);
+    fn state_of_charge(&self, percent: u16, capacity: u16, full_capacity: u16, error: ReturnCode);
+    fn voltage_current(&self, voltage: u16, current: u16, error: ReturnCode);
+    fn coulomb(&self, coulomb: u16, error: ReturnCode);
+    fn romid(&self, error: ReturnCode);
 }
 
 
 pub struct MAX17205<'a> {
     i2c0: &'a i2c::I2CDevice,
+    i2c1: &'a i2c::I2CDevice,
```

Comment:
It would be nice to name these two `i2c` fields in some way that made it clear which kinds of commands can go through each.